src/wallet/rpcwallet.cpp

```diff
@@ -2931,8 +2932,14 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
         addrType = params[0].get_str();
     }
 
-    KeyIO keyIO(Params());
+    const auto& chainparams = Params();
+    KeyIO keyIO(chainparams);
     if (addrType == ADDR_TYPE_SPROUT) {
+        int nextBlockHeight = chainActive.Height() + 1-1;
```

Comment:
why `-1`?
```suggestion
        int nextBlockHeight = chainActive.Height() + 1;
```

Reply:
Oh, actually, why not just use chainActive.Height()? Why use `nextBlockHeight`, we want Sprout address generation to be disabled as of Canopy activation; we don't need to disable it a block early, I don't think.

Reply:
Discussed at standup, we'll disable Sprout address generation one block before Canopy activation since such an address can't be useful since the earliest block it could be used is the next block, at which time it will be disallowed. Several existing wallet RPCs have similar logic. I'll fix that `-1` error (this was for mutation testing and I accidentally left it in).